internal/command/cliconfig/cliconfig.go

```diff
@@ -220,19 +222,50 @@ func loadConfigDir(path string) (*Config, tfdiags.Diagnostics) {
 // Any values specified in this config should override those set in the
 // configuration file.
 func EnvConfig() *Config {
+	env := makeEnvMap(os.Environ())
+	return envConfig(env)
+}
+
+func envConfig(env map[string]string) *Config {
 	config := &Config{}
 
-	if envPluginCacheDir := os.Getenv(pluginCacheDirEnvVar); envPluginCacheDir != "" {
+	if envPluginCacheDir := env[pluginCacheDirEnvVar]; envPluginCacheDir != "" {
 		// No Expandenv here, because expanding environment variables inside
 		// an environment variable would be strange and seems unnecessary.
 		// (User can expand variables into the value while setting it using
 		// standard shell features.)
 		config.PluginCacheDir = envPluginCacheDir
 	}
 
+	if envMayBreak := env[pluginCacheMayBreakLockFileEnvVar]; envMayBreak != "" && envMayBreak != "0" {
+		// This is an environment variable analog to the
+		// plugin_cache_may_break_dependency_lock_file setting. If either this
+		// or the config file setting are enabled then it's enabled; there is
+		// no way to override back to false if either location sets this to
+		// true.
+		config.PluginCacheMayBreakDependencyLockFile = true
+	}
+
 	return config
 }
 
+// getEnv is a copy of
```

Comment:
This comment trails off, and

Reply:
heh and it isn't even a docstring for the function I actually ended up writing :man_facepalming: